content/post/review-building-static-analysis-tools.md

```diff
@@ -0,0 +1,279 @@
+---
+author: alex
+date: 2018-07-26
+title: "Paper review: “Lessons from Building Static Analysis Tools at Google”."
+image: /post/review-building-static-analysis-tools/lessons.png
+description: "Review of a recent scientific paper by Google on the experience of building large-scale static analysis tools."
+categories: ["science", "technical"]
+draft: true
+---
+
+A recent paper with empirical research on the application of static code analysis tools caught my attention:
+
+{{% tweet 988527557300568065 %}}
+
+It’s been a while since I started reading more papers, following awesome initiatives like [Papers We Love](https://paperswelove.org/)
+and [The Morning Paper](https://blog.acolyer.org/) — I always wanted to experiment with publishing notes.
+This will be the first attempt.
+
+{{% center %}} … {{% /center %}}
+
+[“Lessons from Building Static Analysis Tools at Google](https://cacm.acm.org/magazines/2018/4/226371-lessons-from-building-static-analysis-tools-at-google/fulltext)”
+by Caitlin Sadowski, Edward Aftandilian, [Alex Eagle](https://twitter.com/jakeherringbone), Liam Miller-Cushon, Ciera Jaspan
+presents 2 stories: the history of failed attempts at integrating  [FindBugs](https://github.com/findbugsproject/findbugs),
+a static analysis tool for Java at [Google](http://twitter.com/Google), and lessons learned from the success
+story of incorporating extensible analysis framework, [Tricorder](https://research.google.com/pubs/pub43322.html),
+to development workflow at Google.
+
```

Comment:
Do you have an image to illustrate one or both of these stories ?

Reply:
not really